### PR TITLE
CodeJam-3 CoreJS

### PR DESCRIPTION
**Pull request by VictoriaTheBrave**
There are three implemented functions for CodeJam-3 CoreJS as was required. Tests and eslint/eslint-config-airbnb-base packages are added. The code satisfies the eslint-config-airbnb-base requirements.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+.lock-wscript
+
+# Compiled binary addons (http://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# Typescript v1 declaration files
```
Comment (OlgaMardvilko) [inline .gitignore:39]: Are you sure that you need to specify so much for this project?

File: src/sumOfOther.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = function sumOfOther(array) {
+  if (array.length === 0) return [];
```
Comment (andrei-shelenhouski) [inline src/sumOfOther.js:2]: Could you pls use only one 'return'?

File: src/make.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module.exports = function make(...args) {
+  if (typeof args[0] === 'function') {
+    return { TypeError: "Can't call function make with empty array as argument" };
+  }
+  const argumentsArray = [].slice.call(args);
+  function func(...arg) {
+    for (let i = 0; i < arg.length; i += 1) {
```
Comment (andrei-shelenhouski) [inline src/make.js:7]: Use i++

File: src/make.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module.exports = function make(...args) {
+  if (typeof args[0] === 'function') {
+    return { TypeError: "Can't call function make with empty array as argument" };
+  }
+  const argumentsArray = [].slice.call(args);
+  function func(...arg) {
```
Comment (andrei-shelenhouski) [inline src/make.js:6]: 'arg' means only one?

File: .gitignore (hunk 1 above)
Comment (VictoriaTheBrave) [reply .gitignore:39]: If you ask about what is written in .gitignore, I just copied this from our tasks from stage-1. In fact I didn't push to this repository only node_modules folder but this folder was nesessary to meet the requirements of the task such as tests and eslint and so on

File: src/make.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+module.exports = function make(...args) {
```
Comment (andrei-shelenhouski) [inline src/make.js:1]: What about (15, 10, 10, sum) as arguments?

File: src/recursion.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+module.exports = function recursion(tree) {
+  if (tree === undefined) {
+    return undefined;
+  }
+  const result = [[tree.value]];
+
+  const left = recursion(tree.left);
+  if (left !== undefined) {
+    for (let i = 0; i < left.length; i += 1) {
```
Comment (andrei-shelenhouski) [inline src/recursion.js:9]: Use i++

File: src/recursion.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      result.push(left[i]);
+    }
+  }
+
+  const right = recursion(tree.right);
+  if (right !== undefined) {
+    for (let i = 0; i < right.length; i += 1) {
```
Comment (andrei-shelenhouski) [inline src/recursion.js:16]: SAA

File: .gitignore (hunk 1 above)
Comment (OlgaMardvilko) [reply .gitignore:39]: This is not an error and does not affect the assessment of the task. I just wanted to pay attention to the fact that you need to understand what you are doing and why. Ok, I see that you understand the origin of this file. Well done!

Comment (VictoriaTheBrave) [reply .gitignore:39]: Thanks :)

**Review by andrei-shelenhouski — COMMENTED** [review]
sumOfOther = 10
make = 20
recursion = 20
git = 20
package.json/linter = 20
tests = 20

*110*
=====

File: src/make.js (hunk 3 above)
Comment (VictoriaTheBrave) [reply src/make.js:7]: I used. But eslint thinks different so I did what it required

File: src/make.js (hunk 4 above)
Comment (VictoriaTheBrave) [reply src/make.js:6]: No, not one. At first I used empty parentheses and code worked but there was a problem with eslint so I rewrote this way

File: src/make.js (hunk 3 above)
Comment (andrei-shelenhouski) [reply src/make.js:7]: @VictoriaTheBrave I know about eslint. ))) it did not affect the assessment

Comment (VictoriaTheBrave) [reply src/make.js:7]: Ok, just a little explanation))) In future I will use i++ whatever eslint says)))

Comment (andrei-shelenhouski) [reply src/make.js:7]: @VictoriaTheBrave just add it as an exception into .eslintrc.js

Comment (VictoriaTheBrave) [reply src/make.js:7]: Understood, thanks!

File: src/make.js (hunk 5 above)
Comment (VictoriaTheBrave) [reply src/make.js:1]: Thank you, I've fixed it)

**Pull request merged**